### ISSUE TITLE
Having trouble saving minidumps with bugchecks in LXCore

### ISSUE THREAD

**Issue by fpqc** [Microsoft/WSL#1564]
I had a bugcheck recently using WSL, but when I checked the C:\Windows\minidump directory, and C:\Windows\Memory.DMP, I couldn't find them to submit to secure@microsoft.com

I'd really like to submit these dumps to improve the project, but they aren't saving properly.  Are there logs anywhere that I can use to troubleshoot why my memdumps aren't saving? 

This isn't necessarily a WSL problem, but since it's crucial to get the data to the team (and since some bugchecks are intermittent and hard to reproduce), could someone help me out? I tried reading the documentation, and everything seems fine. All the memdump settings are correct, and I only have one pagefile, size managed by Windows, on C:.

The main Windows event viewer only shows an unexpected shutdown at the time it occurred, with no other details about which logs were saved where or whatever.

It does have this message attached under the details tab (doesn't seem like much to go on):

```
- System 

  - Provider 

   [ Name]  Microsoft-Windows-Kernel-Power 
   [ Guid]  {331C3B3A-2005-44C2-AC5E-77220C37D6B4} 
 
   EventID 41 
 
   Version 4 
 
   Level 1 
 
   Task 63 
 
   Opcode 0 
 
   Keywords 0x8000400000000002 
 
  - TimeCreated 

   [ SystemTime]  2017-01-01T10:34:23.587306400Z 
 
   EventRecordID 3567 
 
   Correlation 
 
  - Execution 

   [ ProcessID]  4 
   [ ThreadID]  8 
 
   Channel System 
 
   Computer FPQC-PC 
 
  - Security 

   [ UserID]  S-1-5-18 
 

- EventData 

  BugcheckCode 0 
  BugcheckParameter1 0x0 
  BugcheckParameter2 0x0 
  BugcheckParameter3 0x0 
  BugcheckParameter4 0x0 
  SleepInProgress 0 
  PowerButtonTimestamp 0 
  BootAppStatus 0 
  Checkpoint 0 
  ConnectedStandbyInProgress false 
  SystemSleepTransitionsToOn 0 
  CsEntryScenarioInstanceId 0 
```

**Comment by sunilmut** [comment]
Thanks, @fpqc. We appreciate all of your help with the project.

Your event viewer seems to be consistent with your observation of the dump file. If there was a dump file created, you will see a bug check event (Event ID: 1001?) in the event log, which will look something like: `The computer has rebooted from a bugcheck. The bugcheck was: <bugcheck code> (parameters)`.

Can you double check your dump settings under "Startup and Recovery" [Configure the dump type](https://support.microsoft.com/en-us/kb/254649)? I know that you said above that `All the memdump settings are correct`, but I wasn't sure which settings you were referring to.

You can try to force the bugcheck by following [this](https://msdn.microsoft.com/en-us/library/windows/hardware/ff545499(v=vs.85).aspx) and see if the dump appears?

**Comment by fpqc** [comment]
Excellent, thanks @sunilmut I'll try some other configurations and see if any of them properly dump memory.

**Comment by sunilmut** [comment]
What if you change the "Write debugging information" to "Automatic memory dump", "Kernel memory dump" or "Complete memory dump"? Still no dump files?

**Comment by fpqc** [comment]
@sunilmut I tried it with complete memory dump before and it didn't work.  I can't imagine it will work with the others, but if you want, I can try again later.

**Comment by sunilmut** [comment]
Yes, please. If it still doesn't work, I will try to follow up internally with the team responsible for the dump file generation. I am sure they would want to have this information as well.

**Comment by fpqc** [comment]
@sunilmut Just tried Automatic and Kernel as well, none of them saved dump files.

**Comment by sunilmut** [comment]
Thanks for the confirmation. Let me follow up with the right team (after a repro on my local system). I am assuming you are on 14986? If not, please let me know.

**Comment by fpqc** [comment]
@sunilmut Yeah I'm on 14986. By the way, I do have one last message that might be of import:

After each of the critical failures (inducing the manual crash), I do have a regular error that says: 

`Audit events have been dropped by the transport.  0`

**Comment by sunilmut** [comment]
@fpqc - Thanks. Had a few words with the owning team. Few more questions. Are you seeing any errors in the event log of the form `Crashdump initialization failed` or `Configuring the page file for crash dump failed`. These should be under the source `volmgr` with event ID 46, 49. I am assuming not since you have scanned the events already, but just in case.

**Comment by fpqc** [comment]
@sunilmut I have one event of type 49 from 12/8, but that's the last time. None of my recent tests or the original bugcheck (1/1) had such an event.

**Comment by sunilmut** [comment]
Thanks @fpqc. And, just to be sure, you are also not seeing any error event with id 161 from `volmgr`? I am sure you are not, but I am instructed to ask. I tried to setup a repro on my VM running 14986, but the key sequence to bugcheck the system is not working. Are you trying this on a physical machine?

**Comment by fpqc** [comment]
@sunilmut No events of id161.  And yes, I'm running the fast ring on (all of my) physical hardware (like all courageous heroes).

**Comment by ionescu007** [comment]
Do you have at least 25GB of free disk space? Although you should still see the minidump...

There's also the possibility that this isn't actually a true bug check. Is it possible you're seeing something that says "HARD ERROR" or "CRITICAL PROCESS TERMINATED"?

**Comment by sunilmut** [comment]
@ionescu007 - The smaller dump types should not need 25GB of memory. Also, the [manually initiated crash](https://msdn.microsoft.com/en-us/library/windows/hardware/ff545499(v=vs.85).aspx) should lead to a bugcheck as the article says. I am going to try this on a physical machine running the same build to get a repro.

**Comment by fpqc** [comment]
@ionescu007 I do. The only thing I can think of is I transferred and extended this partition from a much smaller ssd (100G partition on a 200G SSD to 450G on a  900G to much larger one, and the partition is now more full than the original disk was sized (~200G/450G).  Maybe whatever is dumping is not checking the updated partition size?

**Comment by fpqc** [comment]
@sunilmut By the way, could it be because I enabled long filename support in group policy?

**Comment by ionescu007** [comment]
That would normally be a strange bug -- the kernel crashes through direct volume manager access, not file names. You may want to try enabling support for a 'dedicated dump file' to see if that changes anything.

**Comment by fpqc** [comment]
@ionescu007 Yep. Making a dedicated dump file worked.  I am now getting minidumps and full dumps. I picked an old drive that I have sitting empty on my PC (it has bad sectors and is ancient, but I figure it's around if I need to use it in a pinch).  

@sunilmut So the question for the memdump team is why does it think that my C: drive is full when I have 300GB of free space?  The only possible reason I could imagine is that, as I said, I have resized this partition from a 100GB partition on my initial installation to a 450GB partition.
